### DISCUSSION TITLE
repo setup

### DISCUSSION THREAD

**Discussion by sub1120** [lmsGenie/lms-app#1] [Ideas]
@shivamvijaywargi  we can put our suggestions and resources here about monorepo setup.

**Comment by shivamvijaywargi** [comment]
https://medium.com/@guysenpai89/nx-monorepo-publish-libraries-to-github-packages-with-github-actions-semantic-release-970883f2786a#:~:text=Your%20monorepo%20is%20commitizen%2Dfriendly,It%20also%20supports%20commitlint%20config.

**Comment by shivamvijaywargi** [comment]
https://blog.devgenius.io/automatic-semantic-versioning-and-commit-messages-semantic-release-and-commitizen-72e4eca5f194

**Comment by sub1120** [comment]
@shivamvijaywargi, I think you are right, monorepo does not have much use for us, and kind of overhead.

Can we move to other approach separate repo for frontend and backend? 

If we both are good with separate repo approach,

Lets convert the current repo for frontend (because we already have issues created).
And we will make another repo for backend.

**Comment by shivamvijaywargi** [comment]
https://github.com/googleapis/release-please#setting-up-release-please

Please look at this and see if it is beneficial for us. If yes, can we use it?

**Reply by sub1120** [reply]
Ya, its good for release and version management, we can use it.

**Reply by shivamvijaywargi** [reply]
Please setup if possible, also rewrite the GH Action now that we have moved away from NX.

Thank you.

**Reply by sub1120** [reply]
sure I will check it.

**Comment by shivamvijaywargi** [comment]
The client setup has been completed from my end, please check if we are missing something or not.

**Comment by sub1120** [comment]
- Updated Branch protection rule (now owners also will not be able to push to the main branch accidentally)
- Added feature and bug issue template
